dispatcher.c

```diff
@@ -279,37 +280,75 @@ static inline ssize_t
 lzread(z_strm *strm, void *buf, size_t sze)
 {
 	char *ibuf = strm->ibuf;
+	size_t srcsize, destsize, total;
 	int ret;
 
 	/* read any available data, if it fits */
 	ret = strm->nextstrm->strmread(strm->nextstrm,
 			ibuf + strm->ipos, METRIC_BUFSIZ - strm->ipos);
+
+	/* if EOF(0) or no-data(-1) then only get out now if the input buffer is empty
+	   because start of next frame may be waiting for us */
+	
 	if (ret > 0) {
 		strm->ipos += ret;
 	} else if (ret < 0) {
-		return -1;
+		if(strm->ipos == 0)
```

Comment:
similar style-nit: use space after if/for/while

Reply:
done